tests/foreman/virtwho/test_cli_virtwho.py

```diff
@@ -40,45 +39,41 @@
 from robottelo.virtwho_utils import VIRTWHO_SYSCONFIG
 
 
-class VirtWhoConfigTestCase(CLITestCase):
-    @classmethod
-    @skip_if_not_set('virtwho')
-    @skip_if(settings.virtwho.hypervisor_type == 'kubevirt' and is_open('BZ:1735540'))
-    def setUpClass(cls):
-        super(VirtWhoConfigTestCase, cls).setUpClass()
-        cls.satellite_url = settings.server.hostname
-        cls.satellite_username = settings.server.admin_username
-        cls.satellite_password = settings.server.admin_password
-        cls.hypervisor_type = settings.virtwho.hypervisor_type
-        cls.hypervisor_server = settings.virtwho.hypervisor_server
-        cls.hypervisor_username = settings.virtwho.hypervisor_username
-        cls.hypervisor_password = settings.virtwho.hypervisor_password
-        cls.hypervisor_config_file = settings.virtwho.hypervisor_config_file
-        cls.vdc_physical = settings.virtwho.sku_vdc_physical
-        cls.vdc_virtual = settings.virtwho.sku_vdc_virtual
-
-    def _make_virtwho_configure(self):
-        args = {
-            'debug': 1,
-            'interval': '60',
-            'hypervisor-id': 'hostname',
-            'hypervisor-type': self.hypervisor_type,
-            'hypervisor-server': self.hypervisor_server,
-            'organization-id': 1,
-            'filtering-mode': 'none',
-            'satellite-url': self.satellite_url,
-        }
-        if self.hypervisor_type == 'libvirt':
-            args['hypervisor-username'] = self.hypervisor_username
-        elif self.hypervisor_type == 'kubevirt':
-            args['kubeconfig'] = self.hypervisor_config_file
-        else:
-            args['hypervisor-username'] = self.hypervisor_username
-            args['hypervisor-password'] = self.hypervisor_password
-        return args
-
+@fixture()
```

Comment:
Consider module scoping this fixture, as its not changing with the functions.

Reply:
Data returned by this fixture is used to create virt-who config entity, and tests remove said entity and assert it doesn't exist by searching for name (generated here). So we need this fixture to return different data for each function, or run tests sequentially.

Reply:
I see, gen_string is providing some randomization there. Still getting used to the fact that we're executing this parallelization by function across the same target Sat instance, thanks @mirekdlugosz

Reply:
Sorry for the delayed reply as we had the labor holiday, thanks @mirekdlugosz for the detailed explanation.